modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BJSONType.java

```diff
@@ -28,6 +29,8 @@
  */
 public class BJSONType extends BType {
 
+    protected StructDef constraint;
```

Comment:
All BType instances are singleton for the VM. So we can't maintain the scheme type along with the BType. I think this has to go to VarRef + BJSON (value class)